### DISCUSSION TITLE
mix: renovate will open PRs for updates that are not possible

### DISCUSSION THREAD

**Discussion by hschaeidt** [renovatebot/renovate#25088] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I am currently setting up renovate in an Elixir project using the Phoenix framework.

Everything is working fine so far, I am getting PRs for outdated dependencies. However, recently there was a release for the `phoenix_live_view` package that bumped it from 0.19 to 0.20, and renovate opened a pull request for it. So far, so good.

The problem I encounter is that when I run `mix hex.outdated` on the project, this dependency is marked as "Update not possible".

Is this normal behavior, or should renovate not open a PR in this case?

### Relevant debug logs

<details><summary>Logs</summary>

```
$ mix hex.outdated
Dependency              Current  Latest  Status               
ecto_sql                3.10.2   3.10.2  Up-to-date           
esbuild                 0.7.1    0.7.1   Up-to-date           
finch                   0.16.0   0.16.0  Up-to-date           
floki                   0.34.3   0.34.3  Up-to-date           
gettext                 0.20.0   0.23.1  Update possible      
jason                   1.4.1    1.4.1   Up-to-date           
phoenix                 1.7.7    1.7.7   Up-to-date           
phoenix_ecto            4.4.2    4.4.2   Up-to-date           
phoenix_html            3.3.2    3.3.2   Up-to-date           
phoenix_live_dashboard  0.8.2    0.8.2   Up-to-date           
phoenix_live_reload     1.4.1    1.4.1   Up-to-date           
phoenix_live_view       0.19.5   0.20.0  Update not possible
plug_cowboy             2.6.1    2.6.1   Up-to-date           
postgrex                0.17.3   0.17.3  Up-to-date           
swoosh                  1.12.0   1.12.0  Up-to-date           
tailwind                0.2.1    0.2.1   Up-to-date           
telemetry_metrics       0.6.1    0.6.1   Up-to-date           
telemetry_poller        1.0.0    1.0.0   Up-to-date           

Run `mix hex.outdated APP` to see requirements for a specific dependency.

To view the diffs in each available update, visit:
https://hex.pm/l/HMR88
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by viceice** [comment, marked as answer]
this is normal behavior, renovate doesn't have enough knowledge that this update is blocked by something else.

**Reply by rarkins** [reply]
I've been thinking that artifacts functions could be enriched to classify the type of artifacts errors which occur (eg incompatibility, permissions, etc) and then we can give the option to suppress all or selective PRs with artifacts errors. Eg this would allow users to say "don't raise incompatible PRs" and we can achieve that assuming there's an artifacts function which errors

**Comment by hschaeidt** [comment]
Thank you. I will label it accordingly that it will need additional investigation before being merged.
